docs: update create plugin docs (#1953)
medusajs/medusa: docs/content/advanced/backend/plugins/create.md

npm run start
```

> ### Testing Troubleshooting

Reviewer: Change title to "Troubleshoot Errors"

>  If you are certain your plugin is structured correctly and you encounter `error: The class must be a valid service implementation` then you can try the following:

Reviewer: To make this clearer, I think it's best to do the following here:

1. Add a subtitle under "Testing Troubleshooting" saying "Error: The class must be a valid service implementation"
2. Change the current text here to first mention something like "Please make sure that your plugin is following the correct structure. If the error persists then please try the following fix".

```bash npm2yarn

> cd /your/medusa-backend/node_modules/medusa-interfaces

Reviewer: Replace all occurrences of `/your/medusa-backend` with `<SERVER_PATH>`

npm link

> cd /your/plugin/directory

Reviewer: Replace all occurrences of `/your/plugin/directory` with `<PLUGIN_PATH>`

rm -rf node_modules/medusa-interfaces
npm link medusa-interfaces
npm link
cd /your/medusa-backend
npm link your-plugin
```

>  This will link the `medusa-interfaces` package from your `medusa-backend` to your plugin directory and then link your plugin to your `medusa-backend` and should fix the error.

Reviewer: Add before this a message to explain that `<SERVER_PATH>` and `<PLUGIN_PATH>` should be replaced with their actual values. It can be something like "Where `<SERVER_PATH>` is the path to your Medusa server and `<PLUGIN_PATH>` is the path to your plugin."

> :::note

Reviewer: I don't think it's necessary to add this note here so you can remove it.

npm run start
```

### Troubleshoot Errors

#### Error: The class must be a valid service implementation

> Please make sure that your plugin is following the correct structure. If the error persists then please try the following fix.

Reviewer: ```suggestion
Please make sure that your plugin is following the correct structure. If the error persists then please try the following fix:
```

```bash npm2yarn
cd <SERVER_PATH>/node_modules/medusa-interfaces
npm link
cd <PLUGIN_PATH>
rm -rf node_modules/medusa-interfaces
npm link medusa-interfaces
npm link
cd <SERVER_PATH>
npm link your-plugin
```

> Where <SERVER_PATH> is the path to your Medusa server and <PLUGIN_PATH> is the path to your plugin.

Reviewer: ```suggestion
Where `<SERVER_PATH>` is the path to your Medusa server and `<PLUGIN_PATH>` is the path to your plugin.
```

> This will link the `medusa-interfaces` package from your `medusa-backend` to your plugin directory and then link your plugin to your `medusa-backend` and should fix the error.

Reviewer: ```suggestion
This links the `medusa-interfaces` package from your `medusa-backend` to your plugin directory and then links your plugin to your `medusa-backend`.
```

#### APIs not loading

If the APIs you added to your Medussa server are not loading then please try the following steps:

```bash npm2yarn
cd <PLUGIN_PATH>
rm -rf node_modules
cd <SERVER_PATH>/node_modules/<PLUGIN_NAME>
npm install
cd <PLUGIN_PATH>
npm build
cd <SERVER_PATH>
npm start
```

> Where <SERVER_PATH> is the path to your Medusa server, <PLUGIN_PATH> is the path to your plugin and <PLUGIN_NAME> is the name of your plugin as it is in your plugin `package.json` file.

Reviewer: ```suggestion
Where `<SERVER_PATH>` is the path to your Medusa server, `<PLUGIN_PATH>` is the path to your plugin and `<PLUGIN_NAME>` is the name of your plugin as it is in your plugin `package.json` file.
```
